GameProject/Assets/Turn based assets/Scripts/TileGenerator.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TileGenerator : MonoBehaviour
+{
+
+    [SerializeField] GameObject[] tiles;
+    [SerializeField] private int width;
+    [SerializeField] private int length;
+    public List<GameObject> tileList;
+
+    public void Start()
+    {
+        GenerateTiles();
+    }
+
+    public void GenerateTiles()
```

Comment:
Nitpick, but this could be `private` seeing as you've called it from `Start()`, which is good.